src/cps/loader.cpp

```diff
@@ -261,33 +279,44 @@ namespace cps::loader {
 
     Platform::Platform() = default;
 
-    Package::Package() = default;
-    Package::Package(std::string _name, std::string _cps_version,
-                     std::unordered_map<std::string, Component> && _components, std::string cps_path_,
-                     std::optional<std::vector<std::string>> && _default_comps, Requires req,
-                     std::optional<std::string> ver, version::Schema schema)
-        : name{std::move(_name)}, cps_version{std::move(_cps_version)}, components{std::move(_components)},
-          cps_path{std::move(cps_path_)}, default_components{std::move(_default_comps)}, require{std::move(req)},
-          version{std::move(ver)}, version_schema{schema} {};
-
-    tl::expected<Package, std::string> load(const fs::path & path) {
-        std::ifstream file;
-        file.open(path);
-
+    tl::expected<Package, std::string> load(std::istream & input_buffer, std::string const & filename) {
         Json::Value root;
-        file >> root;
+        try {
+            input_buffer >> root;
+        } catch (std::exception const & ex) {
+            return tl::make_unexpected(
+                fmt::format("Exception caught while parsing json for `{}.cps`\n{}", filename, ex.what()));
```

Comment:
We'll want to take another pass at user-friendly diagnostics in a future PR I expect, but let's not hold up this PR on that. I expect a parsing error should look a lot more like a useful compiler error at some point. Like "expected a closing quote on `foobar.cps:42,41`" for instance.